puffin/src/merge.rs

```diff
@@ -8,8 +8,8 @@ struct MergeNode<'s> {
     /// All these scopes have the same `id`.
     pieces: Vec<MergePiece<'s>>,
 
-    /// indexed by their id
-    children: BTreeMap<&'s str, MergeNode<'s>>,
+    /// indexed by their id+data
+    children: BTreeMap<(&'s str, &'s str), MergeNode<'s>>,
```

Comment:
Since the key is getting slightly more complicated, it might help for future maintenance to use a small data structure containing and documenting those two strings here, maybe?

Reply:
that's a good idea, esp. as we already had a `MergeNode` here. otherwise I think the basic comment may have sufficed.

Reply:
done!